l10n_be_iso20022_pain/models/account_move.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2021 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import re
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+def check_bbacomm(val):
+    supported_chars = "0-9+*/ "
+    pattern = re.compile("[^" + supported_chars + "]")
+    if pattern.findall(val or ""):
+        return False
+    bbacomm = re.sub(r"\D", "", val or "")
+    if len(bbacomm) == 12:
+        base = int(bbacomm[:10])
+        mod = base % 97 or 97
+        if mod == int(bbacomm[-2:]):
+            return True
+    return False
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    reference_type = fields.Selection(
+        selection_add=[("bba", "BBA Structured Communication")],
```

Comment:
"bba" in payment order should be "BBA" (uppercase)

Reply:
@ThomasBinsfeld can you elaborate why you downvoted the comment above?
@luc-demeyer if you have a link to the spec of what must go in a SEPA Credit Transfer file for BBA structured communications, that would be helpful?

Reply:
I didn't see the need to change that and imply some refactoring. But this part has been removed since.